.github/workflows/build-push-tool-images.yml

```diff
@@ -0,0 +1,66 @@
+name: Build & Push Tool Image to GHCR
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+    paths:
+      - '.github/workflows/build-push-tool-images.yml'
+      - 'src/loaders/compute_tools/**/Dockerfile'
+  push:
+    branches:
+      - main
+      - master
+      - develop
+    paths:
+      - '.github/workflows/build-push-tool-images.yml'
+      - 'src/loaders/compute_tools/**/Dockerfile'
+
+jobs:
+  build-push-tool-images:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out GitHub Repo
+        uses: actions/checkout@v3
+
+      - name: Get current date
+        id: date
+        run: echo "date=$(date -u +'%Y-%m-%dT%H:%M:%SZ')" >> $GITHUB_OUTPUT
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to GHCR
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: '${{ secrets.GHCR_USERNAME }}'
+          password: '${{ secrets.GHCR_TOKEN }}'
+
+      - name: Build and push tool images
+        id: build-and-push
+        run: |
+          tools=(
+            mash
+            gtdb_tk
+            checkm2
+            microtrait)
+
+          for tool in "${tools[@]}"; do
+            dockerfile="./src/loaders/compute_tools/$tool/Dockerfile"
+            tag="ghcr.io/${{ github.event.repository.owner.login }}/${{ github.event.repository.name }}:$tool"
```

Comment:
Unless I'm mistaken this tag will be the same every time there's an new image, right? So there's no way to select older images?

Reply:
yea, tag will be the same for every image. Setting tags to be collections:tool_name is preferred by sysadmin.
https://kbase.slack.com/archives/C02MHNMCQ/p1689612602761919

But if you have suggestions, I can change how to tag forms. Will there be a case where we need to run tools with old builds?

Reply:
I didn't get the impression that they were stating a preference, just providing options, and the same tag is the easiest.

Anyway, to me having the same tag for all images seems like a debugging / reproducibility nightmare. I'd either incorporate the hash or the version into the tag, so something like mash_<githash> or match_v.0.4.1

Reply:
how do you want the tag to be tracked across the repo? it needs to be pull by task runner.